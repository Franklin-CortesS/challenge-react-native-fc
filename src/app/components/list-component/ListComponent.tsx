
import React, { useState, useEffect} from "react";
import { ActivityIndicator, FlatList, StyleSheet, Text, View } from 'react-native';
import { Avatar, ListItem } from '@react-native-elements/base'

const ListComponent = () => {
    const [isLoading, setLoading] = useState(true);
    const [data, setData] = useState([]);
    const test_image: string = "https://d5nunyagcicgy.cloudfront.net/external_assets/hero_examples/hair_beach_v391182663/original.jpeg";
    const api_url: string = "https://6172cfe5110a740017222e2b.mockapi.io/elements";

    async function getData(){
        try {
            const response = await fetch(api_url);
            const json = await response.json();
            setData(json);
        } catch (error) {
            console.error(error);
        } finally {
            setLoading(false);
        }
    }

    const myListEmpty = () => {
        return (
          <View style={{ alignItems: "center" }}>
          <Text style={styles.item}>No hay usuarios que mostrar.</Text>
          </View>
        );
    };

    useEffect(() => {
        getData();
    }, []);

    return (
        <View style={styles.container}>
        {isLoading ? <ActivityIndicator/> : (
            <FlatList
              data={data}
              keyExtractor={({ id }, index) => id}
              ListEmptyComponent={myListEmpty}
              renderItem={({ item }: any) => (
                  <ListItem key={item.id} bottomDivider>
                    <Avatar source={{uri: test_image}} rounded></Avatar>
                    <ListItem.Content style={styles.listItem}>
                      <ListItem.Title style={styles.listName}>
                        {item.name}
                      </ListItem.Title>
                    </ListItem.Content>
                  </ListItem>
              )}
            />
        )}
      </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1
      },
      item: {
        marginTop: 12,
        fontWeight: 'bold',
        color: "#000000"
      },
      title: {
        fontSize: 32,
      },
      listItem: {
        display: "flex"
      },
      listName: {
        display: "flex"
      }
  });

export default ListComponent;