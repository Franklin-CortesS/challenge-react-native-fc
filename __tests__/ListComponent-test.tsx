/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 
 // Note: test renderer must be required after react-native.
 import renderer from 'react-test-renderer';
 import ListComponent from '../src/app/components/list-component/ListComponent';
 
 describe('Test para Home Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ListComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
 