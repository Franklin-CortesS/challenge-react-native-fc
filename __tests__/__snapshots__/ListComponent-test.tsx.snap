// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para Home Component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ActivityIndicator />
</View>
`;
